common/src/main/java/io/netty/util/internal/NativeLibraryLoader.java

```diff
@@ -371,4 +390,62 @@ private static void closeQuietly(Closeable c) {
     private NativeLibraryLoader() {
         // Utility
     }
+
+    private static final class NoexecVolumeDetector {
+
+        private static boolean canExecuteExecutable(File file)
+                throws ClassNotFoundException, NoSuchMethodException, InvocationTargetException,
+                       IllegalAccessException {
+            if (PlatformDependent.javaVersion() < 7) {
+                // Assume true because we
+                return true;
+            }
+
+            // If we can already execute, there is nothing to do.
+            if (file.canExecute()) {
+                return true;
+            }
+
+            // On volumes, with noexec set, even files with the executable POSIX permissions will fail to execute.
+            // So, check for executable permissions, then find out if the file can be executed.
+
+            // Everything is done reflectively to prevent the animal-sniffer plugin from complainin.
+            Class<?> filesClass = Class.forName("java.nio.file.Files");
```

Comment:
First, if I'm reading, `io.netty.util.internal.PlatformDependent#newConcurrentDeque` correctly, it _appears_ that it is fine to use Java7 classes provided that they are protected by a version check -- no?

Whether or not this is true, I would be happy to do this non-reflectively, but doing so would require adding the classes to the Animal Sniffer configuration.  My understanding of how Animal Sniffer works is that doing so would suppress the usages of these classes globally within the Netty codebase.  Given how natural it has been for developers to use `java.nio.file.*`, this seems unwise.

As of [Animal Sniffer 1.11](http://blog.soebes.de/blog/2014/03/25/animal-sniffer-version-1-dot-11-released/), it is possible to configure an annotation to suppress warnings in specific classes or method, but so far as I can tell, this is not yet configured for the project.  If you want, I could make a separate issue that extended `io.netty.util.SuppressForbidden` or added something similar (`io.netty.util.IgnoreJDK7` and `io.netty.util.IgnoreJDK8`?), added these to relevant scopes, and removed the global class exclusions from the sniffer configuration.

Please advise.